blocks/math/include/gnuradio-4.0/math/Math.hpp

```diff
@@ -115,6 +125,221 @@ using Multiply = MathOpMultiPortImpl<T, std::multiplies<T>>;
+            // |v| == v  ➜ avoid ambiguous std::abs and silence -Wabsolute-value
+            return v;
+        } else if constexpr (std::is_integral_v<T>) {
+            // std::abs promotes to int/long long; cast back to T
+            using Promoted = std::conditional_t<(sizeof(long long) > sizeof(int) &&
+                                                 sizeof(T) > sizeof(int)),
+                                                long long,
+                                                int>;
+            return static_cast<T>(std::abs(static_cast<Promoted>(v)));
+        } else {
+            // float, double, long double: return type is already correct
+            return std::abs(v);
+        }
+    }
+};
+
+template<typename T>
+using Abs = MathOpSinglePortImpl<T, abs_op<T>>;
+
+/* ------------------------------------------------------------------ *
+ *  Integrate : running sum over N samples, output 1 value, decimate   *
+ * ------------------------------------------------------------------ */
+
+ GR_REGISTER_BLOCK("gr::blocks::math::Integrate",
+                  gr::blocks::math::Integrate,
+                  ([T]),
+                  [ uint8_t, uint16_t, uint32_t, uint64_t,
+                    int8_t,  int16_t,  int32_t,  int64_t,
+                    float, double ])
+
+template<typename T>
+requires std::is_arithmetic_v<T>
+struct Integrate : gr::Block<Integrate<T>>
+{
+    /* ---------- Doc string ---------------------------------------- */
+    using Description = Doc<R""(
+@brief Integrate successive samples and decimate.
+
+For every *decim* input items the block outputs the sum of those
+*decim* items and then resets the accumulator.
+
+Example, decim = 4  
+in : 1 2 3 4 5 6 7 8   →  out : 10 26
+)"">;
+
+    /* ---------- Ports & settings ---------------------------------- */
+    PortIn<T>  in;
+    PortOut<T> out;
+
+    Annotated<gr::Size_t,
+              "decim",
+              Visible,
+              Doc<"decimation / integration length">,
+              Limits<1UZ, (1UZ << 20)> >    // ← space before last '>'
+        decim = 1UZ;
+
+    GR_MAKE_REFLECTABLE(Integrate, in, out, decim);
+
+    /* ---------- State --------------------------------------------- */
+    T           _acc  = T(0);
+    std::size_t _seen = 0;
+
+    /* ---------- Reset on start / stop ----------------------------- */
+    void start()        { _acc = T(0); _seen = 0; }
+    /* no explicit stop() needed – Block takes care of it */
+
+    /* ---------- Work (scalar & SIMD transparent) ------------------ */
+    template<gr::InputSpanLike TSpanIn, gr::OutputSpanLike TSpanOut>
+    gr::work::Status processBulk(const TSpanIn& ins, TSpanOut& outs)
+    {
+        auto       outIt   = outs.begin();
+        const auto dec     = decim.value;
+
+        for (auto v : ins) {
+            _acc += v;
+            if (++_seen == dec) {
+                *outIt++ = _acc;
+                _acc  = T(0);
+                _seen = 0;
+            }
+        }
```

Comment:
I think that you can use a `chunk_view` (https://en.cppreference.com/w/cpp/ranges/chunk_view.html) and `std::accumulate` to iterate over this more cleanly.